### DISCUSSION TITLE
Multiple OR Reconstruction - Is It Possible?

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#1149] [Ask Anything] [answered]
Hi everyone,

Has anyone tried employing multiple explicit orientation relationships in a reconstruction process? I have a system that switches between two known orientation relationships, and I was looking to reconstruct the prior parent grains with both of them. I discovered this dual OR behavior when the reconstruction process could only recompute ~ 50% of the microstructure when employing one OR.

My initial thought was to break the reconstruction into two parts:
1) Recalculate everything possible with OR1.
2) Recalculate the remaining material with OR2 and combine this with OR1 for form a fully reconstructed microstructure.

The concern I have with this approach is that the reconstruction process from OR1 may interfere with OR2 (removing the surrounding material may be problematic).

My second thought was to employ a for loop to see which OR better fits a given voxel (by minimizing misorientation of the specified planes and directions) and then operate according to this output. **I could see this taking a long time to compute however, and wanted to see if the community had more insight on a different approach?**

In addition to the explicit OR process, I've also tried the graph approach as documented for martensite reconstructions. This resulted in worse reconstructions than the explicit OR approach, and steered my query here.

Thanks in advance! The reconstruction process in MTEX is amazing and I can't wait to apply it to other systems.
Alec

**Comment by nyyssont** [comment, marked as answer]
Thank you for your thoughts. I'm sure many others (myself included) would be interested in something like this. I think the ORTools add-on for MTEX has some methods already for OR study:
https://github.com/ORTools4MTEX

Regarding your two-part approach, I think it would be as simple as stating:

```
%After initial round of reconstruction:
job.p2c = p2c_desired;
%Continue with reconstruction:
```

On your second thought. How would you test for fitness of the OR? I'd say the true fit in each case would be given by a comparison of the actual parent orientation and the measured child orientation versus your representative OR. If the parent orientation is, for example, that of a fully recrystallized austenite grain, does that match the prior orientations assigned to each individual child grain? Looking at the example for martensite in the documentation, it doesn't look like it. Any thoughts on this?

In any case, I think it's definitely worth it to pursue your idea. Speed will probably not be an issue if it is a case of comparing lists of orientations or ORs. There is usually a way around using loops.

**Comment by Savillian** [comment]
Thank you for the prompt reply, and also thank you for pointing out ORTools exists! This turned out to be the key to dealing with multiple ORs, as the additional functions can automatically (and accurately) determine the ORs intrinsic to a given set of EBSD data. It's also an incredibly fast operation, and I'm very surprised this wasn't obviously listed on the MTEX website. @ralfHielscher could this library be added to the main website?

And regarding the fitness of an OR, I would have implemented something to evaluate angular misorientation between the parent and child phases. If the misorientation in the data differs from that expected for a given OR, I would know to use a different OR for the reconstruction process of this orientation. ORTools already implements this in a streamlined manner and can identify these OR for you to boot however, so this capability already exists.

Kudos to the team behind ORTools, I'm very impressed by the capabilities of this function library.
